### ISSUE TITLE
Chapter 10, exercise 10: Wrong accuracy from method `evaluate` called on best model

### ISSUE THREAD

**Issue by ibeauregard** [ageron/handson-ml2#258]
If you run the code for exercise 10, towards the end, you will get:

```python
model = keras.models.load_model("my_mnist_model.h5") # rollback to best model
model.evaluate(X_test, y_test)
```
Ouput
```
313/313 [==============================] - 0s 1ms/step - loss: 0.0616 - accuracy: 0.0977
[0.061620958149433136, 0.09769999980926514]
```
So apparently, we have 0.0977 accuracy.

However...
```python
y_predicted = np.argmax(model.predict(X_test), axis=-1)
np.count_nonzero(y_predicted == y_test) / len(y_test)
```
Output
```
0.9825
```

Curiously enough, **before assigning the best model to variable `model`** (as in the first code block above), if you run:
```python
model.evaluate(X_test, y_test)
```
You get the following (expected) output:
```
313/313 [==============================] - 0s 1ms/step - loss: 0.0751 - accuracy: 0.9845
[0.0750538781285286, 0.984499990940094]
```

So method `evaluate` works as expected in this situation.

Alternatively, since the best model was saved after 6 epochs, you can refit with only 6 epochs. Then, immediately after training, if you do:

```python
model.evaluate(X_test, y_test)
````
You get the following output:
```
313/313 [==============================] - 0s 1ms/step - loss: 0.0616 - accuracy: 0.9825
[0.061620958149433136, 0.9825000166893005]
```

Note that here the accuracy is the same as the one manually calculated earlier on the best model, which should be equivalent to the model trained with only 6 epochs. So everything is fine here.

Can someone explain what is going on, here?
Thank you!

**Comment by ibeauregard** [comment]
I opened an issue with the TensorFlow repo: https://github.com/tensorflow/tensorflow/issues/43457.
